src/core/public/chrome/nav_links/nav_links_service.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { sortBy } from 'lodash';
+import { BehaviorSubject, ReplaySubject } from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
+import { NavLinkWrapper, NavLinkUpdateableFields } from './nav_link';
+import { ApplicationStart } from '../../application';
+
+interface StartDeps {
+  application: ApplicationStart;
+}
+
+export class NavLinksService {
+  private readonly stop$ = new ReplaySubject(1);
+
+  public start({ application }: StartDeps) {
+    const navLinks$ = new BehaviorSubject<ReadonlyArray<NavLinkWrapper>>(
```

Comment:
I think that makes sense as well, though a `Map` will make more sense than a `Set` since these are keyed by `id`.

Reply:
I feel like this service could benefit from storing the nav links in a Set rather than an array. This would simplify the implementation of `get`, `exists`, and others. In fact, if this service could mimic the operations of Set, that would be nice, e.g. rename `exists` to `has`. Thoughts?